src/snap/sdk.ts

```diff
@@ -160,10 +160,10 @@ async function runKeygen() {
 
 async function runBackup() {
 	let { pairingData, silentShareStorage } = await getPairingDataAndStorage();
-	const encryptedMessage = await encMessage(
-		JSON.stringify(silentShareStorage.newPairingState?.distributedKey),
-	);
-	await Backup.backup(pairingData, encryptedMessage);
+	if (silentShareStorage.newPairingState?.distributedKey) {
+		const encryptedMessage = await encMessage(JSON.stringify(silentShareStorage.newPairingState?.distributedKey));
```

Comment:
we should catch error from encMessage, in this case runBackup should fail

Reply:
Yes runBackup will fail if encMessage fail, dApp will catch the error directly.